A96-retry-otel-stats.md

```diff
@@ -0,0 +1,95 @@
+## A96: OTel Metrics for Retries
+
+*   Author: Yash Tibrewal (@yashykt)
+*   Approver: Mark Roth (@markdroth)
+*   Status: In Review
+*   Implemented in:
+*   Last updated: 2025-04-02
+*   Discussion at: https://groups.google.com/g/grpc-io/c/bFUHkcBA9cw
+
+## Abstract
+
+Propose OpenTelemetry metrics for gRPC retries.
+
+## Background
+
+Since OpenCensus has been sunsetted in favor of OpenTelemetry, gRPC has been
+developing its own OpenTelemetry plugin that is meant to replace the OpenCensus
+plugin ([A66]). This document proposes the OpenTelemetry version of the retry
+metrics originally proposed in [A45].
+
+### Related Proposals:
+
+*   [A45]: Exposing OpenCensus Metrics and Tracing for gRPC retry
+*   [A66]: OpenTelemetry Metrics
+*   [A79]: Non-per-call Metrics Architecture
+
+[A45]: A45-retry-stats.md
+[A66]: A66-otel-stats.md
+[A79]: A79-non-per-call-metrics-architecture.md
+
+## Proposal
+
+Metric Name                  | Type      | Unit    | Labels                                                                     | Description
+---------------------------- | --------- | ------- | -------------------------------------------------------------------------- | -----------
+grpc.client.call.retries     | Histogram | {retry} | grpc.method (required), grpc.target (required), grpc.retry_type (required) | Number of retry attempts made during the client call. The original attempt is not counted as a retry attempt. Recommended histogram bucket boundaries are [0,1,2,3,4,5,10,100,1000].
```

Comment:
And what does it mean for the `retry_type` to be on the `retries` metric with unit `retry`?  What should be reported if there is one transparent retry attempt and two non-transparent retries?  What if they happen in a different order?

Reply:
Outdated. Now, we are just adding it as an attribute on `grpc.client.attempt.started`

Reply:
Now, we have separate metrics for retries, hedges and transparent retries.